helix: Fix `Vim::NextWordEnd` off-by-one in `HelixSelect`

AidanV opened the pull request:
Closes #43209
Closes #38121

Starting on the first character.
Running `v e` before changes: 
<img width="410" height="162" alt="image" src="https://github.com/user-attachments/assets/ee13fa29-826c-45c0-9ea0-a598cc8e781a" />

Running `v e` after changes:
<img width="483" height="166" alt="image" src="https://github.com/user-attachments/assets/24791a07-97df-47cd-9ef2-171522adb796" />

Change Notes:

- Added helix selection sanitation code that directly mirrors the code in the Vim [`visual_motion`](https://github.com/AidanV/zed/blob/b6728c080c5d14ded7002d0276deb5c19d42ed8a/crates/vim/src/visual.rs#L237) method. I kept the comments from the Vim section that explains its purpose.
- The above change converted the problem from fixing `v e` to fixing `v w`. Since `w` is treated differently in Helix than in Vim (i.e. `w` in Vim goes to the first character of a word and `w` in Helix goes to the character before a word. Commented [here](https://github.com/AidanV/zed/blob/b6728c080c5d14ded7002d0276deb5c19d42ed8a/crates/vim/src/helix.rs#L132)), the code treats `w` in `HelixSelect` as a motion that differs from the Vim motion in the same way that the function [`helix_move_cursor`](https://github.com/AidanV/zed/blob/b6728c080c5d14ded7002d0276deb5c19d42ed8a/crates/vim/src/helix.rs#L353) separates these behaviors. 
- Added a regression test

Release Notes:

- Fixes bug where `Vim::NextWordEnd` in `HelixSelect` would not select whole word.

AidanV committed:
helix: Fix NextWordEnd off-by-one

jrobsonchase reviewed (approved):
Diagnosis and fix make sense to me, tested locally and seems to be working as expected :+1: 

This also fixes https://github.com/zed-industries/zed/issues/38121, which seems to be caused by a similar current character/cursor position issue.

AidanV committed:
helix: Test full cursor selection

AidanV commented:
Thanks for pointing that out! I updated the description to also close #38121. 

I added a test that checks the core behavior of the selection being the full width of the cursor. I believe this is testing the same issue outlined in #38121. I did not add the specific test written in #38121 because `escape` clears the selection in Zed but it does not in Helix. If we fixed this later down the line it would cause a false regression.

AidanV committed:
Merge branch 'main' into helix-next-word-end
